src/momento/cache_operation_types.py

```diff
@@ -163,6 +163,124 @@ def values_as_bytes(self) -> List[Optional[bytes]]:
+            return None
+        return cast(bytes, self._value).decode("utf-8")
+
+    def value_as_bytes(self) -> Optional[bytes]:
+        if self.status() == CacheGetStatus.MISS:
+            return None
+        return cast(bytes, self._value)
+
+    def status(self) -> CacheGetStatus:
+        return self._result
+
+    def __str__(self) -> str:
+        return self.__repr__()
+
+    def __repr__(self) -> str:
+        value = self._value
+        try:
+            value = self.value()
+        except UnicodeDecodeError:
+            value = str(value)
+
+        return f"CacheDictionaryGetResponse(value={value}, result={self._result})"
+
+
+class CacheDictionaryValue:
+    def __init__(self, value: DictionaryValue) -> None:
+        self._value = value
+
+    def value(self) -> str:
+        return cast(bytes, self._value).decode("utf-8")
+
+    def value_as_bytes(self) -> bytes:
+        return cast(bytes, self._value)
+
+    def __eq__(self, other: object) -> bool:
+        return type(other) == type(self) and self._value == other._value  # type: ignore
+
+    def __str__(self) -> str:
+        return self.__repr__()
+
+    def __repr__(self) -> str:
+        value = self._value
+        try:
+            value = self.value()
+        except UnicodeDecodeError:
+            pass
+
+        return f"CacheDictionaryValue(value={value})"  # type: ignore
+
+
+# Represents the type of a hash as stored in the cache.
+# This is the type returned by hgetall and hset.
+StoredDictionary = Dict[DictionaryKey, CacheDictionaryValue]
+
+
+class CacheDictionarySetResponse:
+    def __init__(self, key: bytes, value: StoredDictionary):
+        self._key = key
+        self._value = value
+
+    def value(self) -> StoredDictionary:
+        return self._value
+
+    def key(self) -> str:
+        """Decodes key of item set in cache to a utf-8 string."""
+        return self._key.decode("utf-8")
+
+    def key_as_bytes(self) -> bytes:
+        """Returns key of item stored in cache as bytes."""
+        return self._key
+
+    def __str__(self) -> str:
+        return self.__repr__()
+
+    def __repr__(self) -> str:
+        try:
+            key = self.key()
+        except UnicodeDecodeError:
+            key = self.key_as_bytes()  # type: ignore
+
+        return f"CacheDictionarySetResponse(key={key}, value={self._value})"
+
+
+class CacheDictionaryGetAllResponse:
+    def __init__(self, value: Optional[StoredDictionary], result: CacheGetStatus):
+        self._value = value
+        self._result = result
+
+    def value(self) -> Optional[StoredDictionary]:
+        if self.status() != CacheGetStatus.HIT:
+            return None
+        return self._value
+
+    def status(self) -> CacheGetStatus:
+        return self._result
+
+    def __str__(self) -> str:
+        return self.__repr__()
+
+    def __repr__(self) -> str:
+        return (
+            f"CacheDictionaryGetAllResponse(value={self._value}, result={self._result})"
+        )
```

Comment:
Agreed it's odd -- this was done by the black code formatter. The line with the return statement + string is too long (per the settings in black). It's wrapping the string in parens to make it an expression (as opposed to a singleton tuple "(x,)") so it can split it into separate lines. I made some housekeeping issues to revisit this.